specs/waku/waku-rpc-api.md

```diff
@@ -0,0 +1,410 @@
+---
+title: Waku RPC API
+version: 1.0.0
+status: Draft
+authors: Dean Eigenmann <dean@status.im>
+---
+
+## Table of Contents
+
+1. [Abstract](#abstract)
+2. [Wire Protocol](#wire-protocol)
+    1. [Transport](#transport)
+    1. [Objects](#objects)
+    1. [Methods](#methods)
+3. [Copyright](#copyright)
+
+## Abstract
+
+In this specification we describe the RPC API that Waku nodes SHOULD adhere to. The unified API allows clients to easily
+be able to connect to any node implementation.
+
+## Introduction
+
+This API is based off the [Whisper V6 RPC API](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API).
+
+## Wire Protocol
+
+### Transport
+
+Nodes SHOULD expose a [JSON RPC](https://www.jsonrpc.org/specification) API that can be publicly accessed. The JSON RPC version SHOULD be `2.0`. Below is an example request:
+
+```json
+{
+  "jsonrpc":"2.0",
+  "method":"shh_version",
+  "params":[],
+  "id":1
+}
+```
+
+#### Fields
+
+| Field     | Description                                         |
+| --------- | --------------------------------------------------- |
+| `jsonrpc` | Contains the used JSON RPC version (`Default: 2.0`) |
+| `method`  | Contains the JSON RPC method that is being called   |
+| `params`  | An array of parameters for the request              |
+| `id`      | The request ID                                      |
+
+### Objects
+
+In this section you will find objects used throughout the JSON RPC API.
+
+#### Message
+
+The message object represents a Waku message. Below you will find the description of the attributes contained in the message object.
```

Comment:
Perhaps it should be stated what a Waku message is (considering the previous issue with message vs envelope usage).
We could define here the term `Waku message` e.g. as the decrypted payload (and padding) of the Waku envelope + its useful metadata and extra info (such as hash).